markup.go

```diff
@@ -154,26 +150,49 @@ func (m ClassMap) Apply(h *HTML) {
 	Property("className", strings.Join(classes, " ")).Apply(h)
 }
 
-// List represents a list of Markup, Component, or HTML which is individually
+// MarkupList represents a list of Applyer which is individually
 // applied to an HTML element or text node.
-type List []MarkupOrComponentOrHTML
+type MarkupList []Applyer
 
-// Apply implements the Markup interface.
-func (l List) Apply(h *HTML) {
-	for _, m := range l {
-		apply(m, h)
+// Apply implements the Applyer interface.
+func (m MarkupList) Apply(h *HTML) {
+	for _, a := range m {
+		if a == nil {
+			continue
+		}
+		a.Apply(h)
 	}
 }
 
+// Markup wraps a list of Applyer which is individually
+// applied to an HTML element or text node.
+func Markup(m ...Applyer) MarkupList {
+	return MarkupList(m)
+}
+
+// List wraps a list of Markup, Component, or HTML which is individually
+// applied to an HTML element or text node.
+func List(m ...ComponentOrHTML) ChildList {
+	return ChildList(m)
+}
+
 // If returns nil if cond is false, otherwise it returns the given markup.
-func If(cond bool, markup ...MarkupOrComponentOrHTML) MarkupOrComponentOrHTML {
+func If(cond bool, markup ...ComponentOrHTML) ComponentOrHTML {
+	if cond {
+		return ChildList(markup)
+	}
+	return nil
+}
+
+// MarkupIf returns nil if cond is false, otherwise it returns the given markup.
+func MarkupIf(cond bool, markup ...Applyer) Applyer {
```

Comment:
I don't really like having this, but no better options came to mind.

Reply:
I've thought about this for a bit, and haven't come up with any great options either. I'll keep thinking about it.